Added mock-27 (nifH_even) and mock-28 (nifH_tiered)

roey-angel opened the pull request:
fixes #82

nbokulich commented on data/mock-27/dataset-metadata.tsv:
```diff
@@ -0,0 +1,12 @@
+name	nifH_even
```
Please change `nifH_even` to `value` (this is the column heading, not the name of the mock community)

nbokulich commented on data/mock-27/dataset-metadata.tsv:
```diff
@@ -0,0 +1,12 @@
+name	nifH_even
+citation	
```
Please add "NA"... and of course when you have a citation related to this community you can submit a later PR to add this!

nbokulich commented on data/mock-27/dataset-metadata.tsv:
```diff
@@ -0,0 +1,12 @@
+name	nifH_even
+citation	
+qiita-id	NA
+raw-data-url-forward-read	
```
Where are the raw MiSeq reads located? E.g., SRA or EBI urls can be supplied here.

nbokulich commented on data/mock-27/dataset-metadata.tsv:
```diff
@@ -0,0 +1,12 @@
+target-gene	nifH
+target-subfragment	19-473 relative to A. vinelandii nifH (Genbank ACCN# M20568)
+study-type	marker-gene
+sequencing-instrument	Sanger
```
You mentioned MiSeq runs previously... this field should be "MiSeq" rather than the technology used to validate the mock community.

nbokulich commented on data/mock-27/sample-metadata.tsv:
```diff
@@ -0,0 +1,3 @@
+SampleID	BarcodeSequence	LinkerPrimerSequence	ReversePrimer	PrimerName	Description
+Mock.nifH.even	NA			Ueda19F-R6	Mock.nifH.even
```
The barcode sequence should correspond to the MiSeq run that contains these samples.

Also need to add forward and reverse primer sequences here.

nbokulich commented on data/mock-28/dataset-metadata.tsv:
```diff
@@ -0,0 +1,12 @@
+name	nifH_tiered
+citation	
+qiita-id	NA
+raw-data-url-forward-read	
```
need url link to raw MiSeq reads here, also

nbokulich commented on data/mock-28/dataset-metadata.tsv:
```diff
@@ -0,0 +1,12 @@
+target-gene	nifH
+target-subfragment	19-473 relative to A. vinelandii nifH (Genbank ACCN# M20568)
+study-type	marker-gene
+sequencing-instrument	Sanger
```
MiSeq?

nbokulich commented on data/mock-28/dataset-metadata.tsv:
```diff
@@ -0,0 +1,12 @@
+name	nifH_tiered
```
Please change "nifH_tiered" to "value"

nbokulich commented on data/mock-28/sample-metadata.tsv:
```diff
@@ -0,0 +1,3 @@
+SampleID	BarcodeSequence	LinkerPrimerSequence	ReversePrimer	PrimerName	Description
+Mock.nifH.even	NA			Ueda19F-R6	Mock.nifH.even
```
The barcode sequence should correspond to the MiSeq run that contains these samples.

Also need to add forward and reverse primer sequences here.

roey-angel commented:
OK, I've made the suggested changes and pushed them to my repo. Do you see them on the branch?
Comments:
1. For the MiSeq data I've added the SRA url. I assume you then download them to your amazon cloud.
2. I changed to `MiSeq` sample-metadata.tsv , but the sequences given under 'source' are based on Sanger.
3. I've added phylogenetic_classification_CART.tsv which is based on the database from the Zehr lab (probably the most widely used one for the gene). I'd nevertheless like to keep the Silva taxonomy (taxonomy.tsv) because: a. phylogenetic clusters are very broad and not informative enough, b. the mock is based on isolates for which we have 16S data.

nbokulich commented on data/mock-27/sample-metadata.tsv:
```diff
@@ -1,3 +1,2 @@
-Mock.nifH.even	NA			Ueda19F-R6	Mock.nifH.even
-Mock.nifH.tiered	NA			Ueda19F-R6	Mock.nifH.tiered
+SampleID	BarcodeSequence	LinkerPrimerSequence	ForwardPrimer	ReversePrimer	PrimerName	Description
+Mock.nifH.even	GATGACCT	NA	GCIWTYTAYGGIAARGGIGG	GCCATCATYTCICCIGA	Ueda19F-R6	Mock.nifH.even
```
sample names in metadata and taxonomy file(s) must match.

You use `Mock.nifH.even` in the sample metadata but `nifH_even` in taxonomy file(s). Please stick with one in all files for consistency.

roey-angel replied:
done

nbokulich commented on data/mock-27/source/phylogenetic_classification_CART.tsv:
```diff
@@ -0,0 +1,7 @@
+# Phylogenetic classification was done using nifH-CART (https://wwwzehr.pmc.ucsc.edu/CART_model_public/)
+A. torulosa	1	1B
```
is there a header line? I am not sure what the columns mean here. (make sure to use consistent sample names, as above).

roey-angel replied:
I've added the following column names: Species Cluster Sub-cluster

nbokulich commented on data/mock-28/sample-metadata.tsv:
```diff
@@ -1,3 +1,2 @@
-Mock.nifH.even	NA			Ueda19F-R6	Mock.nifH.even
-Mock.nifH.tiered	NA			Ueda19F-R6	Mock.nifH.tiered
+SampleID	BarcodeSequence	LinkerPrimerSequence	ForwardPrimer	ReversePrimer	PrimerName	Description
+Mock.nifH.tiered	TGGTGTTG	NA	GCIWTYTAYGGIAARGGIGG	GCCATCATYTCICCIGA	Ueda19F-R6	Mock.nifH.tiered
```
sample names in metadata and taxonomy file(s) must match.

You use `Mock.nifH.tiered` in the sample metadata but `nifH_tiered` in taxonomy file(s). Please stick with one in all files for consistency.

nbokulich commented on data/mock-28/source/phylogenetic_classification_CART.tsv:
```diff
@@ -0,0 +1,7 @@
+# Phylogenetic classification was done using nifH-CART (https://wwwzehr.pmc.ucsc.edu/CART_model_public/)
+A. torulosa	1	1B
```
is there a header line? I am not sure what the columns mean here. (make sure to use consistent sample names, as above).

nbokulich commented on data/mock-27/dataset-metadata.tsv:
```diff
@@ -1,12 +1,12 @@
 qiita-id	NA
-raw-data-url-forward-read	
-raw-data-url-reverse-read	
+raw-data-url-forward-read	https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR6666663
```
https://s3-us-west-2.amazonaws.com/mockrobiota/latest/mock-27/mock-forward-read.fastq.gz

roey-angel replied:
I changed the url from SRA to this one (assuming that's what you meant here)

nbokulich commented on data/mock-27/dataset-metadata.tsv:
```diff
@@ -1,12 +1,12 @@
-raw-data-url-forward-read	
-raw-data-url-reverse-read	
+raw-data-url-forward-read	https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR6666663
+raw-data-url-reverse-read	https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR6666663
```
https://s3-us-west-2.amazonaws.com/mockrobiota/latest/mock-28/mock-forward-read.fastq.gz

nbokulich commented on data/mock-28/dataset-metadata.tsv:
```diff
@@ -1,12 +1,12 @@
 qiita-id	NA
-raw-data-url-forward-read	
-raw-data-url-reverse-read	
+raw-data-url-forward-read	https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR6666669
```
https://s3-us-west-2.amazonaws.com/mockrobiota/latest/mock-28/mock-forward-read.fastq.gz

nbokulich commented on data/mock-28/dataset-metadata.tsv:
```diff
@@ -1,12 +1,12 @@
-raw-data-url-forward-read	
-raw-data-url-reverse-read	
+raw-data-url-forward-read	https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR6666669
+raw-data-url-reverse-read	https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR6666669
```
https://s3-us-west-2.amazonaws.com/mockrobiota/latest/mock-28/mock-forward-read.fastq.gz

nbokulich commented on data/mock-27/README.md:
```diff
@@ -0,0 +1,23 @@
+
+
+# Known Issues / Notes
+No barcodes, no primers. Sequences are based on Sanger sequencing.
```
You can delete the part about Sanger here, once you add the note above (here and in the mock-28 readme)

nbokulich commented on data/mock-27/README.md:
```diff
@@ -0,0 +1,23 @@
+# mock-nifH-Even
+This mock community was comprised of DNA from the following six diazotrophic species: Anabaena torulosa (Carm.) Lagerh. (Cyanobacteria) Algae Collection Vienna (ASW 01028), Desulfosporosinus acidophilus strain SJ4T (DSM22704) (Firmicutes) (Alazard et al., 2010), Kosakonia sacchari (Gammaproteobacteria) (in house strain, 99.89% sequence identical (1105 bp) to K. sacchari SP1T (Chen et al., 2014)), ...
+Following amplification and purification with ZR-96 DNA Clean-up Kit™ (Zymo) PCR products were quantified using Quant-iT™ PicoGreen® dsDNA Assay Kit (Thermo Fisher Scientific) on a Tecan Safire plate reader (Tecan) and pooled in equal and also tiered proportions.
```
Please add that these were sequenced on a MiSeq (and ideally which sequencing facility, and any other relevant info)

nbokulich replied:
Please add that these were sequenced on a MiSeq (and ideally which sequencing facility, and any other relevant info)

nbokulich commented on data/mock-28/dataset-metadata.tsv:
```diff
@@ -0,0 +1,12 @@
+name	value
+citation	NA
+qiita-id	NA
+raw-data-url-forward-read	https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR6666669
```
https://s3-us-west-2.amazonaws.com/mockrobiota/latest/mock-28/mock-forward-read.fastq.gz

nbokulich commented on data/mock-28/dataset-metadata.tsv:
```diff
@@ -0,0 +1,12 @@
+citation	NA
+qiita-id	NA
+raw-data-url-forward-read	https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR6666669
+raw-data-url-reverse-read	https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR6666669
```
https://s3-us-west-2.amazonaws.com/mockrobiota/latest/mock-28/mock-reverse-read.fastq.gz

nbokulich commented on data/mock-27/README.md:
```diff
@@ -0,0 +1,23 @@
+# mock-nifH-Even
+This mock community was comprised of DNA from the following six diazotrophic species: Anabaena torulosa (Carm.) Lagerh. (Cyanobacteria) Algae Collection Vienna (ASW 01028), Desulfosporosinus acidophilus strain SJ4T (DSM22704) (Firmicutes) (Alazard et al., 2010), Kosakonia sacchari (Gammaproteobacteria) (in house strain, 99.89% sequence identical (1105 bp) to K. sacchari SP1T (Chen et al., 2014)), ...
+Following amplification and purification with ZR-96 DNA Clean-up Kit™ (Zymo) PCR products were quantified using Quant-iT™ PicoGreen® dsDNA Assay Kit (Thermo Fisher Scientific) on a Tecan Safire plate reader (Tecan) and pooled in equal and also tiered proportions.
+
```
Please add an additional paragraph detailing that the taxonomy of all mock community members was validated by Sanger, and these IDs appear in the "source" taxonomy. Please also describe what is contained in the CART file, and what the SILVA taxonomies contain (please add a note that this is not a 16S dataset and the SILVA database will not be useful for classification of these sequences).

Here and in the mock-28 readme

nbokulich commented on data/mock-27/dataset-metadata.tsv:
```diff
@@ -0,0 +1,12 @@
+name	value
+citation	NA
+qiita-id	NA
+raw-data-url-forward-read	https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR6666663
```
https://s3-us-west-2.amazonaws.com/mockrobiota/latest/mock-27/mock-forward-read.fastq.gz

nbokulich commented on data/mock-27/dataset-metadata.tsv:
```diff
@@ -0,0 +1,12 @@
+citation	NA
+qiita-id	NA
+raw-data-url-forward-read	https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR6666663
+raw-data-url-reverse-read	https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR6666663
```
https://s3-us-west-2.amazonaws.com/mockrobiota/latest/mock-27/mock-reverse-read.fastq.gz

nbokulich commented on data/mock-27/sample-metadata.tsv:
```diff
@@ -0,0 +1,2 @@
+SampleID	BarcodeSequence	LinkerPrimerSequence	ForwardPrimer	ReversePrimer	PrimerName	Description
+Mock.nifH.even	GATGACCT	NA	GCIWTYTAYGGIAARGGIGG	GCCATCATYTCICCIGA	Ueda19F-R6	Mock.nifH.even
```
Sample IDs in taxonomy and metadata file must match

nbokulich commented on data/mock-27/source/taxonomy.tsv:
```diff
@@ -0,0 +1,8 @@
+Taxonomy	nifH_even
```
Sample IDs in taxonomy and metadata file must match

nbokulich commented on data/mock-28/README.md:
```diff
@@ -0,0 +1,21 @@
+# mock-nifH-Tiered
+This mock community was comprised of DNA from the following six diazotrophic species: Anabaena torulosa (Carm.) Lagerh. (Cyanobacteria) Algae Collection Vienna (ASW 01028), Desulfosporosinus acidophilus strain SJ4T (DSM22704) (Firmicutes) (Alazard et al., 2010), Kosakonia sacchari (Gammaproteobacteria) (in house strain, 99.89% sequence identical (1105 bp) to K. sacchari SP1T (Chen et al., 2014)), ...
+Following amplification and purification with ZR-96 DNA Clean-up Kit™ (Zymo) PCR products were quantified using Quant-iT™ PicoGreen® dsDNA Assay Kit (Thermo Fisher Scientific) on a Tecan Safire plate reader (Tecan) and pooled in equal and also tiered proportions.
+
```
see notes on mock-27 readme, please add same notes here.

nbokulich commented on data/mock-28/sample-metadata.tsv:
```diff
@@ -0,0 +1,2 @@
+SampleID	BarcodeSequence	LinkerPrimerSequence	ForwardPrimer	ReversePrimer	PrimerName	Description
+Mock.nifH.tiered	TGGTGTTG	NA	GCIWTYTAYGGIAARGGIGG	GCCATCATYTCICCIGA	Ueda19F-R6	Mock.nifH.tiered
```
Sample IDs in taxonomy and metadata file must match

nbokulich commented on data/mock-28/source/taxonomy.tsv:
```diff
@@ -0,0 +1,8 @@
+Taxonomy	nifH_tiered
```
Sample IDs in taxonomy and metadata file must match

roey-angel commented:
I made all changes.
I changed the name of the CART file from "classification" to "clusters" to clarify the idea.
I'm not sure I understand your comment regarding CART. The tool (which is based on the database by Zehr's lab) only returns the phylogenetic cluster of the nifH sequence, not the taxonomic identity of the microbe. In most cases it is impossible to determine the specific taxonomic identity based on nifH sequence.

thermokarst commented:
Hi @roey-angel - did you remember to push those changes to your feature branch? We aren't seeing them reflected in the pull request at the moment. Thanks!

roey-angel commented:
hi @thermokarst, you should see them now
